docs/orchestration.asciidoc

```diff
@@ -0,0 +1,106 @@
+* When an existing NodeSet is removed, ECK migrates data away from the corresponding Elasticsearch nodes to remove, decreases the StatefulSet replicas accordingly, then finally removes the corresponding StatefulSet.
+* When the specification of an existing NodeSet is updated (for example the Elasticsearch configuration, or the PodTemplate resources requirements), ECK performs a rolling upgrade of the corresponding Elasticsearch nodes. In order to do so, it follows link:https://www.elastic.co/guide/en/elasticsearch/reference/current/rolling-upgrades.html[Elasticsearch rolling upgrade best practices], to slowly upgrade Pods to the newest revision while preventing unavailability of the Elasticsearch cluster. In most cases, it corresponds to restarting Elasticsearch nodes one by one, and reusing the same PersistentVolume data. Note that some <<{p}-orchestration-limitations,cluster topologies may cause the cluster to be unavailable during the upgrade>>.
+* When an existing NodeSet is renamed, ECK performs the creation of a new NodeSet with the new name, and the removal of the old NodeSet, according to the NodeSet creation and removal patterns described above. Elasticsearch data is migrated away from the deprecated NodeSet before removal. The Elasticsearch resource <<{p}-update-strategy>> controls how many nodes can exist above or below the target node count during the upgrade.
+
+In every case, ECK ensures any StatefulSet operation is done in a way that is consistent with Elasticsearch orchestration best practices, by manipulating orchestration settings such as `discovery.seed_hosts`, `cluster.initial_master_nodes`, `discovery.zen.minimum_master_nodes`, and `_cluster/voting_config_exclusions` as necessary.
+
+[id="{p}-orchestration-limitations"]
+==== Limitations
+
+Due to the way Kubernetes and StatefulSets behave, ECK orchestration has the following limitations:
+
+* Storage requirements (including volume size) of an existing NodeSet cannot be updated. link:https://github.com/kubernetes/enhancements/issues/661[StatefulSet volumes expansion is not available yet in Kubernetes]. In order to upgrade the storage size, it is however possible to create a new NodeSet, or rename the existing one. Renaming a NodeSet will lead to the creation of a new StatefulSet with the specified storage size. The original StatefulSet is removed once Elasticsearch data has been migrated away to the nodes of the new StatefulSet.
+* Cluster availability cannot be guaranteed during the rolling upgrade of a one-node cluster, or for a cluster having indices with no replicas. If an Elasticsearch node holds the only copy of a shard, this shard becomes unavailable while the node is upgraded. Clusters with more than one node and at least one replica per index are considered best practice.
```

Comment:
s/, or/ or

Reply:
not sure about that one. The comma marks a "pause" in the sentence? (@alaudazzi?)

Reply:
Ah, I'm not sure either :)

Reply:
Either is fine as far as I know, we can also make it a bulleted list to make it clearer